test/models/foreman_git_templates/host_test.rb

```diff
@@ -25,5 +25,78 @@ class HostTest < ActiveSupport::TestCase
         end
       end
     end
+
+    describe '#update' do
+      let(:os) { FactoryBot.create(:operatingsystem, :with_associations, type: 'Redhat') }
+
+      setup do
+        host.expects(:skip_orchestration_for_testing?).at_least_once.returns(false)
+      end
+
+      context 'with valid template_url' do
+        context 'when host is in build mode' do
+          let(:host) { FactoryBot.create(:host, :with_tftp_orchestration, :with_template_url, operatingsystem: os, build: true) }
+
+          it 'updates the host' do
+            ProxyAPI::Resource.any_instance.stubs(:post).returns(true)
+            ProxyAPI::TFTP.any_instance.stubs(:fetch_boot_file).returns(true)
+
+            Dir.mktmpdir do |dir|
+              stub_repository host.params['template_url'], "#{dir}/repo.tar.gz" do |tar|
+                tar.add_file_simple('templates/PXEGrub2/template.erb', 644, host.name.length) { |io| io.write(host.name) }
+              end
+
+              assert host.update(name: 'newname')
+              assert_empty host.errors.messages
+            end
+          end
+        end
+
+        context 'when host is not in build mode' do
+          let(:host) { FactoryBot.create(:host, :with_tftp_orchestration, :with_template_url, operatingsystem: os, build: false) }
+
+          it 'updates the host' do
+            ProxyAPI::TFTP.any_instance.stubs(:set).returns(true)
+
+            Dir.mktmpdir do |dir|
+              stub_repository host.params['template_url'], "#{dir}/repo.tar.gz" do |tar|
+                tar.add_file_simple('templates/PXEGrub2/default_local_boot.erb', 644, host.name.length) { |io| io.write(host.name) }
+              end
+
+              assert host.update(name: 'newname')
+              assert_empty host.errors.messages
+            end
+          end
+        end
+      end
+
+      context 'with invalid template_url' do
+        setup do
+          stub_request(:get, host.params['template_url']).to_return(status: 404)
+        end
+
+        let(:expected_errors) { ["No PXEGrub2 template was found for host #{host.name}. Repository url: #{host.params['template_url']}"] }
```

Comment:
@timogoebel I think this error is expected if the `template_url` is wrong. It's added here: https://github.com/dm-drogeriemarkt/foreman_git_templates/pull/48/files#diff-0a9b35b0c2c90515a22df356a448a6a1R30. So, it should not be possible to update the host with wrong `template_url`.